flare/otf.py

```diff
@@ -83,10 +89,10 @@ def __init__(self, dft_input: str, dt: float, number_of_steps: int,
                  calculate_energy: bool = False, output_name: str = 'otf_run',
                  max_atoms_added: int = 1, freeze_hyps: int = 10,
                  rescale_steps: List[int] = [], rescale_temps: List[int] = [],
-                 dft_softwarename: str = "qe",
-                 no_cpus: int = 1, npool: int = None, mpi: str = "srun",
-                 dft_kwargs=None,
-                 store_dft_output: Tuple[Union[str,List[str]],str] = None):
+                 dft_softwarename: str = "qe", no_cpus: int = 1,
+                 npool: int = None, mpi: str = "srun", dft_kwargs=None,
+                 store_dft_output: Tuple[Union[str, List[str]], str] = None,
+                 custom_module=None):
```

Comment:
How about we re-name "dft_softwarename" to "force_source", and let it take an object or a string as an argument? If the string is 'qe' or 'vasp' or 'cp2k' then it uses the `dft_software` dictionary to link to the correct module, and if it's of type `object`, then it becomes to the `custom_module` case. Would help to cut down on bloat. We could eventually also, with some work, allow for `trajectory` as an option, which could be a first step towards closing #148 .